discovery-provider/src/utils/redis_metrics.py

```diff
@@ -363,13 +419,46 @@ def update_personal_metrics(key, old_timestamp, timestamp, value, metric_type):
         redis_set_and_dump(REDIS, key, json.dumps(updated_metrics))
     logger.info(f"updated cached personal {metric_type} metrics")
 
+def update_summed_unique_metrics(now, ip):
+    thirty_one_days_ago = now - timedelta(days=31)
+    thirty_one_days_ago_str = thirty_one_days_ago.strftime(day_format)
+    yesterday = now - timedelta(days=1)
+    yesterday_str = yesterday.strftime(day_format)
+    today_str = now.strftime(day_format)
+    this_month_str = f"{today_str[:7]}/01"
+
+    summed_unique_daily_metrics_str = redis_get_or_restore(REDIS, summed_unique_daily_metrics)
+    summed_unique_daily_metrics_obj = json.loads(summed_unique_daily_metrics_str) \
+        if summed_unique_daily_metrics_str else {}
+    if today_str not in summed_unique_daily_metrics_obj:
+        summed_unique_daily_metrics_obj[today_str] = [ip]
+    elif ip not in summed_unique_daily_metrics_obj[today_str]:
+        summed_unique_daily_metrics_obj[today_str] = summed_unique_daily_metrics_obj[today_str] + [ip]
+    summed_unique_daily_metrics_obj = {timestamp: ips for timestamp, ips in summed_unique_daily_metrics_obj.items() \
+        if timestamp >= yesterday_str}
+    redis_set_and_dump(REDIS, summed_unique_daily_metrics, json.dumps(summed_unique_daily_metrics_obj))
+
+    summed_unique_monthly_metrics_str = redis_get_or_restore(REDIS, summed_unique_monthly_metrics)
+    summed_unique_monthly_metrics_obj = json.loads(summed_unique_monthly_metrics_str) \
+        if summed_unique_monthly_metrics_str else {}
+    if this_month_str not in summed_unique_monthly_metrics_obj:
```

Comment:
I think we could probably skip actually storing the ips here since we don't need to deduplicate, but this also works :) 

maybe im missing the reason you needed to do it this way

Reply:
it's because we need to know that the ip is unique for that given discovery node